src/backup.ts

```diff
@@ -0,0 +1,505 @@
+    const gaxOpts =
+      typeof gaxOptionsOrCallback === 'object' ? gaxOptionsOrCallback : {};
+    const callback =
+      typeof gaxOptionsOrCallback === 'function' ? gaxOptionsOrCallback : cb!;
+
+    this.bigtable.request<google.protobuf.IEmpty>(
+      {
+        client: 'BigtableTableAdminClient',
+        method: 'deleteBackup',
+        reqOpts: {
+          name: this.name,
+        },
+        gaxOpts,
+      },
+      (err, resp) => callback(err, this, resp)
+    );
+  }
+
+  get(gaxOptions?: CallOptions): Promise<GetBackupResponse>;
+  get(callback: GetBackupCallback): void;
+  get(gaxOptions: CallOptions, callback: GetBackupCallback): void;
+  /**
+   * Get a backup if it exists.
+   *
+   * @param {object} [gaxOptions] Request configuration options, outlined here:
+   *     https://googleapis.github.io/gax-nodejs/CallSettings.html.
+   * @param {function} callback The callback function.
+   * @param {?error} callback.err An error returned while making this
+   *     request.
+   * @param {Backup} callback.backup The Backup instance.
+   * @param {object} callback.apiResponse The full API response.
+   */
+  get(
+    gaxOptionsOrCallback?: CallOptions | GetBackupCallback,
+    cb?: GetBackupCallback
+  ): void | Promise<GetBackupResponse> {
+    const gaxOpts =
+      typeof gaxOptionsOrCallback === 'object' ? gaxOptionsOrCallback : {};
+    const callback =
+      typeof gaxOptionsOrCallback === 'function' ? gaxOptionsOrCallback : cb!;
+
+    this.bigtable.request<IBackup>(
+      {
+        client: 'BigtableTableAdminClient',
+        method: 'getBackup',
+        reqOpts: {
+          name: this.name,
+        },
+        gaxOpts,
+      },
+      (err, resp) => {
+        if (resp) {
+          this.metadata = resp;
+        }
+
+        callback(err, this, this.metadata);
+      }
+    );
+  }
+
+  getMetadata(gaxOptions?: CallOptions): Promise<GetMetadataResponse>;
+  getMetadata(callback: GetMetadataCallback): void;
+  getMetadata(gaxOptions: CallOptions, callback: GetMetadataCallback): void;
+  /**
+   * Get a backup if it exists.
+   *
+   * @param {object} [gaxOptions] Request configuration options, outlined here:
+   *     https://googleapis.github.io/gax-nodejs/CallSettings.html.
+   * @param {function} callback The callback function.
+   * @param {?error} callback.err An error returned while making this
+   *     request.
+   * @param {object} callback.apiResponse The full API response.
+   */
+  getMetadata(
+    gaxOptionsOrCallback?: CallOptions | GetMetadataCallback,
+    cb?: GetMetadataCallback
+  ): void | Promise<GetMetadataResponse> {
+    const gaxOpts =
+      typeof gaxOptionsOrCallback === 'object' ? gaxOptionsOrCallback : {};
+    const callback =
+      typeof gaxOptionsOrCallback === 'function' ? gaxOptionsOrCallback : cb!;
+
+    // @TODO `this.metadata` isn't a `Backup`. Figure out why it has to be cast.
+    this.get(gaxOpts, err => callback(err, this.metadata as Backup));
```

Comment:
I didn't see the suggestion about moving the function body from `getMetadata()` to `get()`. I'm not opposed to it, but why is one better than the other? (I've since made the other corrections to these functions, and they all seem to be co-existing happily)

Reply:
Perhaps didn't click "Add review comment"
My suggestion was to move the function body from `get()` to `getMetadata()` as the snippet above.
Then in `get()` do 
```ts
    this.getMetadata(
      gaxOptions,
      (err, metadata) => {
        callback(err, err ? null : this, metadata);
      }
    );
```
Following an existing example in [Cluster](https://github.com/googleapis/nodejs-bigtable/blob/cb77d569551bab8dd2897c31ea5a47aa0ae99706/src/cluster.ts#L305-L322), [Instance](https://github.com/googleapis/nodejs-bigtable/blob/cb77d569551bab8dd2897c31ea5a47aa0ae99706/src/instance.ts#L620-L635)

Reply:
Gotcha, thanks!